### ISSUE TITLE
Bluemap displays only one claim from XClaim

### ISSUE THREAD

**Issue by Peregrine05** [WasabiThumb/xclaim#106]
This is probably related to #98 and #64. After starting the server (even after clearing the entire `bluemap/web` directory), only one claim is displayed on the map. Usually, the same claim is shown each time, but it does vary occasionally. Reloading Bluemap with `/bluemap reload` does not fix the issue; rather, it causes all claims to disappear from the map. `/xc restart` also does not fix the issue, and instead causes XClaim to crash and stop working entirely, requiring a server restart to restore functionality.

- OS: Linux
- Minecraft version: 1.21
- Purpur version: 1.21-2275
- XClaim version: 1.13.1
- BlueMap version: 5.3-paper

Other plugins:
- DiscordSRV-Build-1.27.0
- EssentialsX-2.20.1
- grimac-2.3.65
- LuckPerms-Bukkit-5.4.137
- squaremap-paper-mc1.21-1.2.6
- TreeFella-1.3
- Vault-1.7.3

**Comment by WasabiThumb** [comment]
I might want to talk to the BlueMap devs again. For #64, IIRC the devs verified that the integration was implemented correctly. It has something to do with ID collisions, even though they're observably unique.

**Comment by Peregrine05** [comment]
After updating a few things, it now seems to be registering all of the claims consistently (at least for the four times I have started the server since making the changes).

The following updates I applied:
- Update to Purpur version: purpur-1.21-2278
- Update to EssentialsX version: EssentialsX-2.21.0-dev+106-8b08a8f
- Install Pl3xMap version: Pl3xMap-1.21-507

The server log is below, if it helps any.
```
server@mc-server:~/mc-server$ ./start.sh
Starting org.bukkit.craftbukkit.Main
[17:14:27 INFO]: [bootstrap] Running Java 21 (OpenJDK 64-Bit Server VM 21.0.4+7-LTS; Eclipse Adoptium Temurin-21.0.4+7) on Linux 6.5.0-45-generic (amd64)
[17:14:27 INFO]: [bootstrap] Loading Purpur 1.21-2278-ver/1.21@b3623fa (2024-07-31T11:13:40Z) for Minecraft 1.21
[17:14:27 INFO]: [PluginInitializerManager] Initializing plugins...
[17:14:28 INFO]: [PluginInitializerManager] Initialized 10 plugins
[17:14:28 INFO]: [PluginInitializerManager] Bukkit plugins (10):
 - BlueMap (5.3), DiscordSRV (1.27.0), Essentials (2.21.0-dev+106-8b08a8f), GrimAC (2.3.65), LuckPerms (5.4.137), Pl3xMap (1.21-507), TreeFella (1.2), Vault (1.7.3-b131), XClaim (1.13.1), squaremap (1.2.6)
[17:14:32 INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, name=PROD]
[17:14:35 INFO]: Loaded 1290 recipes
[17:14:35 INFO]: Loaded 1400 advancements
[17:14:35 INFO]: Starting minecraft server version 1.21
[17:14:35 INFO]: Loading properties
[17:14:35 INFO]: This server is running Purpur version 1.21-2278-ver/1.21@b3623fa (2024-07-31T11:13:40Z) (Implementing API version 1.21-R0.1-SNAPSHOT)
[17:14:36 INFO]: [spark] This server bundles the spark profiler. For more information please visit https://docs.papermc.io/paper/profiling
[17:14:36 INFO]: Server Ping Player Sample Count: 12
[17:14:36 INFO]: Using 4 threads for Netty based IO
[17:14:37 INFO]: [ChunkTaskScheduler] Chunk system is using 1 I/O threads, 3 worker threads, and population gen parallelism of 3 threads
[17:14:38 INFO]: Default game type: SURVIVAL
[17:14:38 INFO]: Generating keypair
[17:14:38 INFO]: Starting Minecraft server on *:25565
[17:14:38 INFO]: Using epoll channel type
[17:14:38 INFO]: Paper: Using libdeflate (Linux x86_64) compression from Velocity.
[17:14:38 INFO]: Paper: Using OpenSSL 3.0.x (Linux x86_64) cipher from Velocity.
[17:14:41 INFO]: [LuckPerms] Loading server plugin LuckPerms v5.4.137
[17:14:41 INFO]: [Vault] Loading server plugin Vault v1.7.3-b131
[17:14:41 INFO]: [Essentials] Loading server plugin Essentials v2.21.0-dev+106-8b08a8f
[17:14:41 INFO]: [BlueMap] Loading server plugin BlueMap v5.3
[17:14:41 INFO]: [GrimAC] Loading server plugin GrimAC v2.3.65
[17:14:42 INFO]: [GrimAC] Loading PacketEvents...
[17:14:47 INFO]: [DiscordSRV] Loading server plugin DiscordSRV v1.27.0
[17:14:47 INFO]: [TreeFella] Loading server plugin TreeFella v1.2
[17:14:47 INFO]: [squaremap] Loading server plugin squaremap v1.2.6
[17:14:47 INFO]: [Pl3xMap] Loading server plugin Pl3xMap v1.21-507
[17:14:47 INFO]: [XC] Loading server plugin XClaim v1.13.1
[17:14:47 INFO]: Server permissions file permissions.yml is empty, ignoring it
[17:14:47 INFO]: [LuckPerms] Enabling LuckPerms v5.4.137
[17:14:48 INFO]:         __    
[17:14:48 INFO]:   |    |__)   LuckPerms v5.4.137
[17:14:48 INFO]:   |___ |      Running on Bukkit - Purpur
[17:14:48 INFO]: 
[17:14:48 INFO]: [LuckPerms] Loading configuration...
[17:14:48 INFO]: [LuckPerms] Loading storage provider... [H2]
[17:14:49 INFO]: [LuckPerms] Loading internal permission managers...
[17:14:49 INFO]: [LuckPerms] Performing initial data load...
[17:14:49 INFO]: [LuckPerms] Successfully enabled. (took 1448ms)
[17:14:49 INFO]: [Vault] Enabling Vault v1.7.3-b131
[17:14:49 INFO]: [Vault] [Economy] Essentials Economy found: Waiting
[17:14:49 INFO]: [Vault] [Permission] SuperPermissions loaded as backup permission system.
[17:14:49 INFO]: [Vault] Enabled Version 1.7.3-b131
[17:14:49 INFO]: [LuckPerms] Registered Vault permission & chat hook.
[17:14:49 INFO]: [squaremap] Enabling squaremap v1.2.6
[17:14:53 INFO]: [io.undertow] starting server: Undertow - 2.3.15.Final
[17:14:53 INFO]: [org.xnio] XNIO version 3.8.16.Final
[17:14:53 INFO]: [org.xnio.nio] XNIO NIO Implementation Version 3.8.16.Final
[17:14:53 INFO]: [org.jboss.threads] JBoss Threads version 3.5.0.Final
[17:14:54 INFO]: [squaremap] Internal webserver running on 0.0.0.0:8080
[17:14:54 INFO]: Preparing level "world"
[17:14:55 INFO]: Preparing start region for dimension minecraft:overworld
[17:14:57 INFO]: Time elapsed: 1815 ms
[17:14:57 INFO]: Preparing start region for dimension minecraft:the_nether
[17:14:58 INFO]: Time elapsed: 862 ms
[17:14:58 INFO]: Preparing start region for dimension minecraft:the_end
[17:14:58 INFO]: Time elapsed: 240 ms
[17:14:58 INFO]: [Essentials] Enabling Essentials v2.21.0-dev+106-8b08a8f
[17:14:59 INFO]: [Essentials] Attempting to convert old kits in config.yml to new kits.yml
[17:14:59 INFO]: [Essentials] No kits found to migrate.
[17:15:00 INFO]: [Essentials] Loaded 42679 items from items.json.
[17:15:00 INFO]: [Essentials] Using locale en_US
[17:15:00 INFO]: [Essentials] ServerListPingEvent: Spigot iterator API
[17:15:01 INFO]: [Essentials] Starting Metrics. Opt-out using the global bStats config.
[17:15:01 INFO]: [Vault] [Economy] Essentials Economy hooked.
[17:15:01 INFO]: [Essentials] Using Vault based permissions (LuckPerms)
[17:15:01 INFO]: [BlueMap] Enabling BlueMap v5.3
[17:15:01 INFO]: [BlueMap] Saving all worlds once, to make sure the level.dat is present...
[17:15:01 WARN]: A manual (plugin-induced) save has been detected while server is configured to auto-save. This may affect performance.
[17:15:03 INFO]: [BlueMap] Loading...
[17:15:03 INFO]: [GrimAC] Enabling GrimAC v2.3.65
[17:15:03 INFO]: [GrimAC] Registering singular bukkit event... (PistonEvent)
[17:15:03 INFO]: [GrimAC] Registering packets...
[17:15:03 INFO]: [GrimAC] Registering tick schedulers...
[17:15:03 INFO]: [GrimAC] [ACF] Enabled Asynchronous Tab Completion Support!
[17:15:03 INFO]: [DiscordSRV] Enabling DiscordSRV v1.27.0
[17:15:03 INFO]: [TreeFella] Enabling TreeFella v1.2
[17:15:03 INFO]: [Pl3xMap] Enabling Pl3xMap v1.21-507
[17:15:04 INFO]: [BlueMap] Loading resources...
[17:15:04 INFO]: [DiscordSRV] DiscordSRV is up-to-date. (6405e850eea61a4b9bf4f0d0cbb80fe20b3cfb90)
[17:15:05 INFO]: [BlueMap] Loading textures...
[17:15:06 INFO]: [BlueMap] Baking resources...
[17:15:06 INFO]: [BlueMap] Resources loaded.
[17:15:06 INFO]: [BlueMap] The map 'world' has no dimension configured.
Assuming world: 'world' and dimension: 'minecraft:overworld'.
[17:15:06 INFO]: [BlueMap] Initializing Storage: 'file' (Type: 'bluemap:file')
[17:15:06 INFO]: [BlueMap] Loading map 'world'...
[17:15:07 INFO]: [BlueMap] The map 'world_the_end' has no dimension configured.
Assuming world: 'world_the_end' and dimension: 'minecraft:the_end'.
[17:15:07 INFO]: [BlueMap] Loading map 'world_the_end'...
[17:15:07 INFO]: [BlueMap] The map 'world_nether' has no dimension configured.
Assuming world: 'world_nether' and dimension: 'minecraft:the_nether'.
[17:15:07 INFO]: [BlueMap] Loading map 'world_nether'...
[17:15:07 INFO]: [BlueMap] WebServer bound to all network interfaces on port 8100
[17:15:07 INFO]: [BlueMap] WebServer started.
[17:15:07 INFO]: [BlueMap] Loaded!
[17:15:08 INFO]: [Pl3xMap] [INFO] Internal webserver running on 0.0.0.0:8081
[17:15:08 INFO]: [Pl3xMap] [INFO] Platform: purpur
[17:15:08 INFO]: [Pl3xMap] [INFO] Version: 1.21-507
[17:15:09 INFO]: [DiscordSRV] [JDA] Login Successful!
[17:15:09 INFO]: [XC] Enabling XClaim v1.13.1
[17:15:10 INFO]: [DiscordSRV] [JDA] Connected to WebSocket
[17:15:10 INFO]: [XC] Locating JAR file
[17:15:10 INFO]: [XC] Checking for Dynmap
[17:15:10 INFO]: [XC] Loading trusted players
[17:15:10 INFO]: [XC] Loading claims
[17:15:10 INFO]: [XC] Unpacking claims
[17:15:10 INFO]: [XC] Initializing chunk editor
[17:15:10 INFO]: [XC] Loading command manager
[17:15:10 INFO]: [XC] Registering default commands
[17:15:11 INFO]: [org.reflections.Reflections] Reflections took 314 ms to scan 1 urls, producing 6 keys and 26 values
[17:15:11 INFO]: [DiscordSRV] [JDA] Finished Loading!
[17:15:11 INFO]: [DiscordSRV] Found server G:MC Server(791462592609648691)
[17:15:11 INFO]: [DiscordSRV] - TC:server-information(1093970317287890964)
[17:15:11 INFO]: [DiscordSRV] - TC:game-information(1093944004107841717)
[17:15:11 INFO]: [DiscordSRV] - TC:minecraft-chat-bridge(791462592609648693)
[17:15:11 INFO]: [DiscordSRV] - TC:minecraft-console(1093872817067528272)
[17:15:11 INFO]: [DiscordSRV] Console forwarding assigned to channel TC:minecraft-console(1093872817067528272)
[17:15:11 INFO]: [XC] Starting movement routine
[17:15:11 INFO]: [XC] Refreshing grace routine
[17:15:11 INFO]: [XC] Done
[17:15:11 ERROR]: [STDERR] [com.mysql.jdbc.Driver] Loading class `com.mysql.jdbc.Driver'. This is deprecated. The new driver class is `com.mysql.cj.jdbc.Driver'. The driver is automatically registered via the SPI and manual loading of the driver class is generally unnecessary.
[17:15:11 INFO]: [spark] Starting background profiler...
[17:15:11 INFO]: [DiscordSRV] Enabling Essentials hook
[17:15:11 INFO]: [DiscordSRV] Enabling LuckPerms hook
[17:15:11 INFO]: Done preparing level "world" (17.213s)
[17:15:11 INFO]: Starting remote control listener
[17:15:11 INFO]: Thread RCON Listener started
[17:15:11 INFO]: RCON running on 0.0.0.0:25575
[17:15:11 INFO]: Running delayed init tasks
[17:15:11 INFO]: [Essentials] Essentials found a compatible payment resolution method: Vault Compatibility Layer (v1.7.3-b131)!
[17:15:11 INFO]: [XC] Saving claims
[17:15:11 INFO]: Done (45.624s)! For help, type "help"
[17:15:11 INFO]: [squaremap] Fetching version information...
[17:15:11 INFO]: [Vault] Checking for Updates ... 
[17:15:11 INFO]: [Essentials] Fetching version information...
[17:15:11 INFO]: [Vault] No new version available
[17:15:11 INFO]: [DiscordSRV] Cleared all pre-existing slash commands in 1/1 guilds (0 cancelled)
```

**Comment by Peregrine05** [comment]
I tested it again with the same configuration, and only one claim was registered.

**Comment by WasabiThumb** [comment]
Blue was able to identify a small oversight that I will definitrly fix, but it shouldn't cause this issue. Are you by any chance hot reloading? This isn't fully supported; best results will be from a cold start

**Comment by Peregrine05** [comment]
I have not been hot reloading. For every instance, I have fully `/stop`ped and restarted the server.

**Comment by WasabiThumb** [comment]
I am genuinely stumped. All I can do at this point I guess is to fix that minor issue and completely re-create your setup. Will let you know.

**Comment by Peregrine05** [comment]
The potentially relevant system specs are:
- Xubuntu (Ubuntu 22.04.4 LTS)
- 500 GB HDD
- Intel Xeon X5690
- 12 GB RAM

**Comment by Peregrine05** [comment]
After a recent power outage, the first startup of the server only registered one claim on BlueMap. Then I updated DiscordSRV and EssentialsX, and both times the server has restarted since then have registered all claims successfully, although I doubt that to be due to the plugin updates, and maybe rather to hard disk access speed.

**Comment by RHX0R3** [comment]
Hello,

I can confirm, that this issue is present.
My server also has that problem.

**Comment by WasabiThumb** [comment]
I've made some changes to fix the only oversight Blue was able to identify on the branch [bug/bluemap-hook](https://github.com/WasabiThumb/xclaim/tree/bug/bluemap-hook) and will test it later today

**Comment by RHX0R3** [comment]
Hey there,

maybe compare together with the author of **[Lands](https://www.spigotmc.org/resources/lands-%E2%AD%95-land-claim-plugin-%E2%9C%85-grief-prevention-protection-gui-management-nations-wars-1-21-support.53313/)**, if it can't be resolved.
They also have a Bluemap / Dynmap integration. And maybe they know a thing or two, on how to counter this too.

But, I am confident, that you can figure that out and fix the display error! :)

_Best regards,
X0R3_

**Comment by WasabiThumb** [comment]
Working fine for me on the branch I mentioned. Note that it also worked fine on my server before the changes I made. Since it does not seem to actively cause a regression, I'm merging the changes. However, this isn't a guarantee that it will fix the issue on your server, and I eagerly await your results.
![image](https://github.com/user-attachments/assets/5bb574a2-3d0d-4f8d-888d-4b7fee1e5af9)

[Here](https://send.zcyph.cc/download/7a62e77ef4093380/#J-jwBSLr1Kv6ZC5CB3JfmQ) is the exact JAR I used in this screenshot, on Purpur 1.21

**Comment by Peregrine05** [comment]
All right; I shall test this shortly.

**Comment by Peregrine05** [comment]
Testing three times (the first of the tests was done after a reboot of the computer), all claims are properly registered and displayed every time.

**Comment by RHX0R3** [comment]
Can confirm, that all claims are displayed for now.
Seems like you fixed it! :)

Also;
did you also encounter something like this?

![Unbenannt](https://github.com/user-attachments/assets/37bd65a1-7ad1-4a3b-be5f-6af5fa7fba45)

**Comment by WasabiThumb** [comment]
> Can confirm, that all claims are displayed for now. Seems like you fixed it! :)
> 
> Also; did you also encounter something like this?
> 
> ![Unbenannt](https://private-user-images.githubusercontent.com/37308703/356846995-37bd65a1-7ad1-4a3b-be5f-6af5fa7fba45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMzOTkxOTksIm5iZiI6MTcyMzM5ODg5OSwicGF0aCI6Ii8zNzMwODcwMy8zNTY4NDY5OTUtMzdiZDY1YTEtN2FkMS00YTNiLWJlNWYtNmFmNWZhN2ZiYTQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODExVDE3NTQ1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzZjI4MzkzNDgzZDExMGY1ZTJhOWQ2NjhkNDRiNGYyYjg4MzRhODQwOWU2MTI5YmNhMTM5MWFjOGY1ODc2ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XvnOBJha9lkWZ3NUWJt_Omgx4FHuGmuHOacjSfA1dBo)

No, but I am not surprised. The tracing algorithm has always been kind of questionable and this is why ``old-outline-style`` is still in the config. However, I had to solve essentially the exact same problem (converting a monochrome matrix to a shape) to make [SVG QR Codes](https://github.com/WasabiThumb/enqr/blob/master/src/renderer/impl/svg.ts#L244), and I think that implementation is much better. I might port it over to Java and use it here :)

**Comment by WasabiThumb** [comment]
@RHX0R3 I'll try to sneak in a fix for that visual glitch into [1.14](https://github.com/WasabiThumb/xclaim/pull/121); if I don't get a chance to do that before demand increases then I'll release 1.14 anyways, but I'm hopeful :)

**Comment by WasabiThumb** [comment]
@RHX0R3 Should be fixed! See #125
